New Maestro E2E CI pipeline

janicduplessis opened the pull request:
## What changed (plus any additional context for devs)

This adds a new CI pipeline for Maestro E2E tests on Android.

This currently implements only a few tests as examples, more will be added later.

## Screen recordings / screenshots

N/A

## What to test

N/A

janicduplessis commented on src/components/asset-list/RecyclerAssetList2/index.tsx:
```diff
@@ -189,47 +189,58 @@ function NavbarOverlay({ accentColor, position }: { accentColor?: string; positi
+        </Box>
       </Box>
-      <Box style={{ top: navbarHeight + insets.top, zIndex: 100 }}>
+      <Box
```
Maestro cannot see views that overflow their parent, so I move the navbar outside of the view it used to be in.

jinchung replied:
could you add PoW screen recordings for iOS and Android compared to before/after with these changes

janicduplessis replied:
Ended up reverting this change since there was another change that fixed this (https://github.com/rainbow-me/rainbow/commit/eca35767325337ed3727fc19e6c22f43f0e4ce34).

brunobar79 commented:
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/@janic/e2e-ci/8388cbc1221a01e6b13c952df4989d4bbb1ccb91.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/@janic/e2e-ci/8388cbc1221a01e6b13c952df4989d4bbb1ccb91.ipa) for 8388cbc1221a01e6b13c952df4989d4bbb1ccb91

brunobar79 commented:
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/@janic/e2e-ci/c3810c9eadbfa6674338720b6b965598235c95da.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/@janic/e2e-ci/c3810c9eadbfa6674338720b6b965598235c95da.ipa) for c3810c9eadbfa6674338720b6b965598235c95da

brunobar79 commented:
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/@janic/e2e-ci/37384f5ec250518d31125bde76f5739609628849.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/@janic/e2e-ci/37384f5ec250518d31125bde76f5739609628849.ipa) for 37384f5ec250518d31125bde76f5739609628849

brunobar79 commented:
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/@janic/e2e-ci/15542d4ace2e68d132af0fb95578d7e68d0b48bb.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/@janic/e2e-ci/15542d4ace2e68d132af0fb95578d7e68d0b48bb.ipa) for 15542d4ace2e68d132af0fb95578d7e68d0b48bb

BrodyHughes reviewed (approved):
lgtm! The documentation is helpful and I have already been running the tests locally and it works great.

brunobar79 commented:
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/@janic/e2e-ci/7ddee0f3d597fb1d722d33397585046160c1e140.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/@janic/e2e-ci/7ddee0f3d597fb1d722d33397585046160c1e140.ipa) for 7ddee0f3d597fb1d722d33397585046160c1e140

janicduplessis commented on .github/actions/android-emulator/action.yml:
```diff
@@ -0,0 +1,57 @@
+  api-level:
+    description: 'API level to use for the emulator'
+    required: true
+    default: '30'
```
This is actually the defaults, maybe should remove to avoid confusion? The actual values used are here https://github.com/rainbow-me/rainbow/pull/6478/files#diff-1e0004123e59e5a0dc0c9d0d86eae242c2f02ff983b21de082c7cf840672b0a5R18-R20

BrodyHughes replied:
ahh yeah i see. i would maybe say removing makes sense.

brunobar79 commented:
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/@janic/e2e-ci/1ace6bad83919ece98fc277dfdb4063ebaa34c61.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/@janic/e2e-ci/1ace6bad83919ece98fc277dfdb4063ebaa34c61.ipa) for 1ace6bad83919ece98fc277dfdb4063ebaa34c61

brunobar79 commented:
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/@janic/e2e-ci/83422a49b90595ce8f3c92a514429ada5fab7279.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/@janic/e2e-ci/83422a49b90595ce8f3c92a514429ada5fab7279.ipa) for 83422a49b90595ce8f3c92a514429ada5fab7279

janicduplessis commented:
I am seeing tests fail sometimes because a simulator get the System Ui is not reponding popup that prevents the test from working. For now I will reduce the tests to run on a single emulator at a time. We could also try running on bigger runner, but I think this is ok for now. We could also see if there’s a way to dismiss the popup if it appears, maybe at the start of each test.

brunobar79 commented:
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/@janic/e2e-ci/9c8440153c189e0f7531a653bbac0d3e62045648.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/@janic/e2e-ci/9c8440153c189e0f7531a653bbac0d3e62045648.ipa) for 9c8440153c189e0f7531a653bbac0d3e62045648

jinchung commented on .github/workflows/macstadium-tests.yml:
```diff
@@ -175,5 +175,79 @@ jobs:
+          restore-keys: xcode-cache-deriveddata-${{ github.workflow }}-
+
+      - name: Install Maestro
+        run: curl -fsSL "https://get.maestro.mobile.dev" | bash
```
Maestro releases seem to be put out quite often, can we select a specific version instead (same for the eventual android flow)

jinchung commented on .github/workflows/macstadium-ios-e2e.yml:
```diff
@@ -158,22 +157,32 @@ jobs:
 
-      - name: Detox iOS e2e tests parallel
+      - name: Install Maestro
+        run: curl -fsSL "https://get.maestro.mobile.dev" | bash
```
Maestro releases seem to be put out quite often, can we select a specific version instead (same android flow)

jinchung commented on package.json:
```diff
@@ -275,13 +268,13 @@
     "react-native-video": "6.3.0",
     "react-native-view-shot": "4.0.2",
-    "react-native-vision-camera": "4.4.2",
+    "react-native-vision-camera": "4.6.4",
```
could I get a confirmation of PoW for the camera upgrade on ios and android still working as expected

janicduplessis replied:
<img width="379" alt="image" src="https://github.com/user-attachments/assets/2a5119fb-e918-496b-88a1-a4ec16eb9ea7" />

![IMG_3811](https://github.com/user-attachments/assets/b8094de8-219a-4a7c-bbf3-37397994f1f4)

I don't have a real Android device on hands currently, but was able to test on emulator. I also didn't test scanning an actual QR code as I am not sure exactly how that works. LMK if that's ok.

jinchung commented on src/components/TestDeeplinkHandler.tsx:
```diff
@@ -0,0 +1,43 @@
+import { Linking } from 'react-native';
+import URL from 'url-parse';
+
+export function TestDeeplinkHandler() {
```
could you add a comment here to clarify its purpose - I keep thinking it's something to do with testing deeplink handling; can add a comment even just referring to the line on the e2e/README file regarding it

brunobar79 commented:
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/@janic/e2e-ci/7d57dbca77ec19c5e9123799603a24a2590d8177.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/@janic/e2e-ci/7d57dbca77ec19c5e9123799603a24a2590d8177.ipa) for 7d57dbca77ec19c5e9123799603a24a2590d8177

brunobar79 commented:
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/@janic/e2e-ci/dfa26b26b339d7f639ea36e0d7bcbc38e61451a6.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/@janic/e2e-ci/dfa26b26b339d7f639ea36e0d7bcbc38e61451a6.ipa) for dfa26b26b339d7f639ea36e0d7bcbc38e61451a6

janicduplessis commented:
@BrodyHughes Looks like android build started failing since merging your PR, any ideas what could cause this? I looked at it and don't see why it would cause issues.

It is failing at uploadSentryNativeSymbolsForRelease, which I added some code to avoid including the sentry plugin for e2e test builds. It seems to still work since I added some logs to make sure the sentry gradle plugin is not included.

It is also building fine locally. Just wondering if you see anything that I missed, still investigating now.

janicduplessis commented:
@BrodyHughes Yea it is really weird, I tried disabling the gradle cache, but didn't make a difference. I also just tried to completely remove the sentry gradle plugin code and still doesn't work.

Thanks for having a look!

jinchung commented on ios/Rainbow/AppDelegate.mm:
```diff
@@ -88,9 +88,9 @@ - (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(
   
   BOOL success = [super application:application didFinishLaunchingWithOptions:launchOptions];
-  BOOL isDetox = [[[NSProcessInfo processInfo] arguments] containsObject:@"-IS_TEST"];
+  BOOL isE2E = [[[NSProcessInfo processInfo] arguments] containsObject:@"isE2ETest"];
```
wanted to double check if we need to keep the `-` in front of `isE2ETest` for this?

janicduplessis replied:
Double checked and that works, I guess it was just detox that was adding a `-` in front.

jinchung commented:
let's get this PR merged, and then as a follow up we can remove the last bits of anything related to detox

brunobar79 commented:
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/@janic/e2e-ci/3456a6633eeceaa87c064ce2d7d3101d1b128f2b.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/@janic/e2e-ci/3456a6633eeceaa87c064ce2d7d3101d1b128f2b.ipa) for 3456a6633eeceaa87c064ce2d7d3101d1b128f2b

brunobar79 commented:
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/@janic/e2e-ci/e0d481c4b8387c54b9acb4c9c461ca77de1bd987.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/@janic/e2e-ci/e0d481c4b8387c54b9acb4c9c461ca77de1bd987.ipa) for e0d481c4b8387c54b9acb4c9c461ca77de1bd987

brunobar79 commented:
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/@janic/e2e-ci/b8a938d1ed5231e8339a1831decb4c901c97b55e.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/@janic/e2e-ci/b8a938d1ed5231e8339a1831decb4c901c97b55e.ipa) for b8a938d1ed5231e8339a1831decb4c901c97b55e

brunobar79 commented:
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/@janic/e2e-ci/70c17b630c818749403fbc4d7f8a4569448f57e2.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/@janic/e2e-ci/70c17b630c818749403fbc4d7f8a4569448f57e2.ipa) for 70c17b630c818749403fbc4d7f8a4569448f57e2

brunobar79 commented:
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/@janic/e2e-ci/985c0c8ec70a2f3dc8457811bf4d6bf96157d82c.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/@janic/e2e-ci/985c0c8ec70a2f3dc8457811bf4d6bf96157d82c.ipa) for 985c0c8ec70a2f3dc8457811bf4d6bf96157d82c

brunobar79 commented:
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/@janic/e2e-ci/a2e38a058b3fc2f5d424f3defb0b0c0f3fcf031c.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/@janic/e2e-ci/a2e38a058b3fc2f5d424f3defb0b0c0f3fcf031c.ipa) for a2e38a058b3fc2f5d424f3defb0b0c0f3fcf031c

brunobar79 commented:
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/@janic/e2e-ci/aab8ec2db424eda135e0361fc4bc17302116eee4.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/@janic/e2e-ci/aab8ec2db424eda135e0361fc4bc17302116eee4.ipa) for aab8ec2db424eda135e0361fc4bc17302116eee4

brunobar79 commented:
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/@janic/e2e-ci/75037935b1e9a8fc0b040564545e24c29edf3a5f.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/@janic/e2e-ci/75037935b1e9a8fc0b040564545e24c29edf3a5f.ipa) for 75037935b1e9a8fc0b040564545e24c29edf3a5f

brunobar79 commented:
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/@janic/e2e-ci/1737585e0147c710da988719c7330ddff9311aaf.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/@janic/e2e-ci/1737585e0147c710da988719c7330ddff9311aaf.ipa) for 1737585e0147c710da988719c7330ddff9311aaf

brunobar79 commented:
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/@janic/e2e-ci/545134d471b9f303bfafce6928edafe42a4444d5.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/@janic/e2e-ci/545134d471b9f303bfafce6928edafe42a4444d5.ipa) for 545134d471b9f303bfafce6928edafe42a4444d5

brunobar79 commented:
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/@janic/e2e-ci/1f4a0aafee2bc4580f7b55026647a86d7844ef98.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/@janic/e2e-ci/1f4a0aafee2bc4580f7b55026647a86d7844ef98.ipa) for 1f4a0aafee2bc4580f7b55026647a86d7844ef98

brunobar79 commented:
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/@janic/e2e-ci/67899c7b0aedc90ec9d0221e1981abbbaaeff88e.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/@janic/e2e-ci/67899c7b0aedc90ec9d0221e1981abbbaaeff88e.ipa) for 67899c7b0aedc90ec9d0221e1981abbbaaeff88e

brunobar79 commented:
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/@janic/e2e-ci/9546393ac417603f98976d1f49af1d8663345219.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/@janic/e2e-ci/9546393ac417603f98976d1f49af1d8663345219.ipa) for 9546393ac417603f98976d1f49af1d8663345219

brunobar79 commented:
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/@janic/e2e-ci/17b339fa870b5dbdd40e7fd2e55bb8cfd7f21e59.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/@janic/e2e-ci/17b339fa870b5dbdd40e7fd2e55bb8cfd7f21e59.ipa) for 17b339fa870b5dbdd40e7fd2e55bb8cfd7f21e59

brunobar79 commented:
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/@janic/e2e-ci/9b8e7fe5d131594291b9fae904810f918f558ca1.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/@janic/e2e-ci/9b8e7fe5d131594291b9fae904810f918f558ca1.ipa) for 9b8e7fe5d131594291b9fae904810f918f558ca1

brunobar79 commented:
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/@janic/e2e-ci/6f90ac8ad97abc733e80c9334145fe25b3225f13.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/@janic/e2e-ci/6f90ac8ad97abc733e80c9334145fe25b3225f13.ipa) for 6f90ac8ad97abc733e80c9334145fe25b3225f13

brunobar79 commented:
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/@janic/e2e-ci/f12b8c3eb00047628660c2ffe4f4aa0a0465670f.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/@janic/e2e-ci/f12b8c3eb00047628660c2ffe4f4aa0a0465670f.ipa) for f12b8c3eb00047628660c2ffe4f4aa0a0465670f

jinchung reviewed (approved):
🌮
